### PR TITLE
Add windows installer

### PR DESCRIPTION
**Pull request by kevin-shelaga**
# Description

Add pwsh script for installing glooctl on windows, update docs and add a change log

# Context

Resolves: https://github.com/solo-io/gloo/issues/5062

# Checklist:

- [x] I included a concise, user-facing changelog (for details, see https://github.com/solo-io/go-utils/tree/master/changelogutils) which references the issue that is resolved.
- [x] If I updated APIs (our protos) or helm values, I ran `make -B install-go-tools generated-code` to ensure there will be no code diff
- [x] I followed guidelines laid out in the Gloo Edge [contribution guide](https://docs.solo.io/gloo-edge/latest/contributing/)
- [x] I opened a draft PR or added the work in progress label if my PR is not ready for review
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] I have added tests that prove my fix is effective or that my feature works

BOT NOTES: 
resolves https://github.com/solo-io/gloo/issues/5062

### CODE REVIEW

File: docs/content/getting_started/_index.md [hunk 1]
```diff
@@ -8,9 +8,15 @@ We know you want to jump right in and start hacking away with Gloo Edge. That's
 {{< tabs >}}
 {{< tab name="glooctl" codelang="shell">}}
 # Install glooctl
+
+## Linux/MacOS
 curl -sL https://run.solo.io/gloo/install | sh
 export PATH=$HOME/.gloo/bin:$PATH
 
+## Windows
```
Comment (jenshu) [inline docs/content/getting_started/_index.md:16]: there are a few other places under `docs/` that reference `https://run.solo.io/gloo/install`, should we add the windows instructions there too?

File: docs/content/getting_started/_index.md [hunk 2]
```diff
@@ -8,9 +8,15 @@ We know you want to jump right in and start hacking away with Gloo Edge. That's
...
+curl -sL https://run.solo.io/gloo/windows/install | pwsh
```
Comment (jenshu) [inline docs/content/getting_started/_index.md:17]: where do you get this url? when i curl'd it, nothing was there?

File: projects/gloo/cli/install.ps1 [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+
+if ($glooctlDownloaded -eq $true) {
+
+    Rename-Item -Path $filename -NewName "glooctl.exe"
+    New-Item -ItemType directory -Path "$env:userprofile\.gloo\bin\" -ErrorVariable capturedErrors -ErrorAction SilentlyContinue
+    Move-Item "glooctl.exe" "$env:userprofile\.gloo\bin\" -Force
+    
+    Set-Location (Get-Item $currentLocation).DirectoryName
+    
+    Write-Output "Gloo was successfully installed!"
```
Comment (jenshu) [inline projects/gloo/cli/install.ps1:83]: nit: for this whole section can we update `Gloo` -> `Gloo Edge`?

File: projects/gloo/cli/install.ps1 [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+    Write-Output `n
+    Write-Output "Add the gloo CLI to your path with:"
+    Write-Output '  $env:Path += ";$env:userprofile/.gloo/bin/"'
+    Write-Output `n
+    Write-Output "Now run:"
+    Write-Output "  glooctl install gateway     # install gloo's function gateway functionality into the 'gloo-system' namespace"
+    Write-Output "  glooctl install ingress     # install very basic Kubernetes Ingress support with Gloo into namespace gloo-system"
+    Write-Output "  glooctl install knative     # install Knative serving with Gloo configured as the default cluster ingress"
+    Write-Output "Please see visit the Gloo Installation guides for more:  https://gloo.solo.io/installation/"
```
Comment (jenshu) [inline projects/gloo/cli/install.ps1:92]: i know it redirects, but should we update to the new url `https://docs.solo.io/gloo-edge/latest/installation/`?

File: projects/gloo/cli/install.ps1 (hunk 3 above)
Comment (jenshu) [reply projects/gloo/cli/install.ps1:83]: if you make these changes can you also update the other install script in this folder?

File: docs/content/getting_started/_index.md (hunk 2 above)
Comment (kevin-shelaga) [reply docs/content/getting_started/_index.md:17]: @jenshu ya this url doesnt exist yet, i was hoping we could add a redirect similar to the linux/mac install. It just redirects to github basically

Comment (jenshu) [reply docs/content/getting_started/_index.md:17]: ah ok, do we need to set that up before this PR is merged?

File: docs/content/installation/glooctl_setup.md [hunk 5]
```diff
@@ -15,6 +15,13 @@ The `glooctl` command line provides useful functions to install, configure, and
   export PATH=$HOME/.gloo/bin:$PATH
   ```
 
+* To install on windows you can use this install script. Openssl is required for installtion to execute properly.
```
Comment (jenshu) [inline docs/content/installation/glooctl_setup.md:18]:
```suggestion
* To install on windows you can use this install script. Openssl is required for installation to execute properly.
```

File: docs/content/getting_started/_index.md (hunk 2 above)
Comment (kevin-shelaga) [reply docs/content/getting_started/_index.md:17]: yes, im not sure who we should talk to about this thought :S

**Review by jenshu — APPROVED** [review]

**Pull request merged**